custom_components/frigate/camera.py

```diff
@@ -136,12 +137,12 @@ def __init__(
         # The device_class is used to filter out regular camera entities
         # from motion camera entities on selectors
         self._attr_device_class = DEVICE_CLASS_CAMERA
+        self._stream_source = None
         self._attr_is_streaming = (
             self._camera_config.get("rtmp", {}).get("enabled")
             or self._cam_name
             in self._frigate_config.get("go2rtc", {}).get("streams", {}).keys()
         )
-        self._attr_is_recording = self._camera_config.get("record", {}).get("enabled")
```

Comment:
If I understand correctly, this is supposed to be set by HA when the Stream component is recording. If we want an attribute reflecting whether Frigate is recording it should be a separate one.

Reply:
No that is incorrect, the integration is fully able to set this as it knows if the camera is being recorded as well

Reply:
I'm not sure you're right - HA has its own recording service for cameras (through the Stream component), and specifically cannot record WebRTC streams. From what I understand, this property is supposed to be set by the recording service.
I'm not 100% sure though, so I added it back.

Reply:
it is an attribute that the integration can use to set the state https://developers.home-assistant.io/docs/core/entity/camera/

I would suggest limiting your changes to what is actually relevant for this feature and not removing things that are unrelated